src/dmclock_server.h

```diff
@@ -112,41 +112,47 @@ namespace crimson {
 
 
     struct RequestTag {
-      double reservation;
-      double proportion;
-      double limit;
-      bool   ready; // true when within limit
-      Time   arrival;
+      double   reservation;
+      double   proportion;
+      double   limit;
+      uint64_t cost;
```

Comment:
consider exposing a typedef for `Cost` as part of the interface. this would help disambiguate it from other uint64_t parameters, and calling code would be less likely to rely on its exact representation

Reply:
Very nice idea. Latest version does this.